code/modules/asset_cache/asset_cache_item.dm

```diff
@@ -21,10 +25,20 @@
 	/// TRUE for keeping local asset names when browse_rsc backend is used
 	var/keep_local_name = FALSE
 
-/datum/asset_cache_item/New(name, file)
+///pass in a valid file_hash if you have one to save it from needing to do it again.
+///pass in a valid dmi file path string e.g. "icons/path/to/dmi_file.dmi" to make generating the hash less expensive
+/datum/asset_cache_item/New(name, file, file_hash, dmi_file_path)
 	if (!isfile(file))
 		file = fcopy_rsc(file)
-	hash = md5(file)
+
+	hash = file_hash
+
+	//the given file is directly from a dmi file and is thus in the rsc already, we know that its file_hash will be correct
+	if(!hash)
```

Comment:
doubled check

Reply:
actually whole logic is doubled here, looks like bad copypaste

Reply:
it's definitely not doubled wut